packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -16,6 +16,36 @@ import patch from '../patch'
 import readChunk from '../_readStreamChunk'
 import { generateToken } from '../utils'
 
+// networks field contains:
+// - <n>/ip <=> <n>/ipv4/0
+// - <n>/ipv4/<m>
+// - <n>/ipv6/<m>
+// which n correspond to the PIF
+// for more information, see: https://github.com/xapi-project/xen-api/blob/324bc6ee6664dd915c0bbe57185f1d6243d9ed7e/ocaml/xapi/xapi_guest_agent.ml#L59-L81
+const IPV4_KEY_RE = /^\d+\/ipv4\/\d+$/
+const IPV6_KEY_RE = /^\d+\/ipv6\/\d+$/
+const extractIp = networks => {
+  if (networks === undefined) {
+    return
+  }
+
+  let ipv4, ipv6
+
+  // <min(n)>/ipv4/<min(m)> || <min(n)>/ipv6/<min(m)>
+  for (const key of Object.keys(networks).sort()) {
+    if (IPV4_KEY_RE.test(key)) {
+      ipv4 = networks[key]
+      break
+    }
+
+    if (ipv6 === undefined && IPV6_KEY_RE.test(key)) {
+      ipv6 = networks[key]
+    }
+  }
+
+  return ipv4 ?? ipv6
```

Comment:
You can simplify it a little by returning the IPv4 immediately:
```js
let ipv6
for (...) {
  if (isIpv4) return networks[key]
  if (isIpv6) ipv6 = networks[key]
}
return ipv6
```
Except if you think it's less readable like this.